tests/azuredelete/internal/azure/azureclient.go

```diff
@@ -0,0 +1,61 @@
+package azure
+
+import (
+	"context"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-05-01/resources"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/giantswarm/microerror"
+	"os"
+)
+
+type Client struct {
+	groupsClient *resources.GroupsClient
+}
+
+func NewClient() (*Client, error) {
+	clientID, exists := os.LookupEnv("AZURE_WC_CLIENT_ID")
+	if !exists {
+		return nil, microerror.Maskf(missingEnvVarError, "missing AZURE_WC_CLIENT_ID environment variable")
+	}
+	clientSecret, exists := os.LookupEnv("AZURE_WC_CLIENT_SECRET")
+	if !exists {
+		return nil, microerror.Maskf(missingEnvVarError, "missing AZURE_WC_CLIENT_SECRET environment variable")
+	}
+	tenantID, exists := os.LookupEnv("AZURE_WC_TENANT_ID")
+	if !exists {
+		return nil, microerror.Maskf(missingEnvVarError, "missing AZURE_WC_TENANT_ID environment variable")
+	}
+	subscriptionID, exists := os.LookupEnv("AZURE_WC_SUBSCRIPTION_ID")
+	if !exists {
+		return nil, microerror.Maskf(missingEnvVarError, "missing AZURE_WC_SUBSCRIPTION_ID environment variable")
+	}
+	groupsClient, err := newResourceGroupClient(clientID, clientSecret, tenantID, subscriptionID)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return &Client{
+		groupsClient: groupsClient,
+	}, nil
+}
```

Comment:
Let's talk about this. IIUC @tuommaki suggested using the organization credentials for that cluster to talk to the Azure API, instead of relying on environment variables. Also, if we extract a function that creates an Azure client using the org credentials to a common package, it can be used from several tests. I think I like that approach.

Reply:
Yep, will work on that tomorrow

Reply:
![Brake](https://media.giphy.com/media/9DcvtQ8SiDZm5uOeY3/giphy.gif)

Reply:
this is done BTW